### ISSUE TITLE
CRAN packages using texi2dvi in vignettes/Makefile

### ISSUE THREAD

**Issue by jarioksa** [vegandevs/vegan#158]
Below an email from CRAN:

This concerns packages

HandTill2001 hsdar maSAE vegan

which do as the subject line says.  There are number of problems:

a) texi2dvi need not be present.  It is an optional install on
Unix-alikes, and MiKTeX on Windows removed it in late 2015.

b) It might be present and not work.  The version in texinfo 6.0 uses
bashisms not in POSIX /bin/sh, and that in 6.1 fails when repeated LaTeX
runs are needed (as in hsdar maSAE vegan).

So your usages need correction.  Do consider if the Makefile is needed
at all and whether standard mechanisms could be used.  If it is needed,
a construction like

$(R_HOME)/bin/Rscript -e "tools::texi2pdf('name.tex', clean = TRUE)"

is more portable.  For vegan, which has a .texi file, texi2pdf is
probably needed for that file and should be in SystemRequirements.

-- 
Brian D. Ripley,                  ripley@stats.ox.ac.uk
Emeritus Professor of Applied Statistics, University of Oxford

**Comment by jarioksa** [comment]
An obvious solution is to remove `texi2pdf` dependence. Branch https://github.com/vegandevs/vegan/tree/texi2dvi-removal removes `Makefile` and translates `FAQ-vegan.texi` to `FAQ-vegan.Rmd` and uses **knitr** to build it to a standard `.html` vignette (but not to `.pdf`).  

Changes to previous behaviour are:

* No `NEWS.html` is generated, but we only have `inst/NEWS.Rd`. The `html` was generated by `Makefile` that was now dropped. 
* No `FAQ-vegan.pdf` is generated, but we only have 'inst/doc/FAQ-vegan.html`.

These changes mean that `vegandocs()` must be re-structured or reconsidered. The `vegandocs()` function was originally designed to show non-vignette documentation. Now **R** has developed and all that documentation may be accessible without having a dedicated `vegandocs()` function.

**Comment by jarioksa** [comment]
The following standard **R** commands work:
```r
news(package="vegan")
vignette("FAQ-vegan", package="vegan")
vignette("decision-vegan", package="vegan")
vignette("intro-vegan", package="vegan")
vignette("diversity-vegan", package="vegan")
```
The `NEWS` and  `FAQ` opened in a web browser and other as `pdf` files in my desktop Linux. The vignette names could not be abbreviated. The only thing missing is `partitioning.pdf`. We need to find a solution for that, but otherwise I suggest `vegandocs()` is phased out.

**Comment by jarioksa** [comment]
`texinfo` 6.1 causes problems with **R**. **R** has some tricks to circumvent these problems, but **vegan** fails. The latest `texinfo` is now in Debian, and Kurt Hornik preferred a quick fix. Moving to `.Rmd`will fix this in the main branch, but that conflicts with the `cran-2.3` branch, and I made there a q quick fix that only replaces the `vignettes/FAQ-vegan.texi` with a plain text `inst/FAQ-vegan` and removes `vignettes/Makefile`. The fix is available in https://github.com/vegandevs/vegan/tree/issue-%23158-quick-fix. This also means that we shall have **vegan** 2.3-4. This will be a real minor release: apart from the `.texi` fix it will only have fixes for `linestack` labels, `simper` print and `biodiver` index 19.

**Comment by jarioksa** [comment]
The `texi2dvi` issue and other anomalies were fixed in 3035e70cdc0caaf82ffa6369d. The changes can be ported to cran-2.3 and after cherry-picking, we'll release **vegan**-2.3-4.

**Comment by jarioksa** [comment]
Closed with the v2.3-4 release.
